packages/ng/libraries/core/src/lib/user/select/homonyms/user-homonyms.service.ts

```diff
@@ -0,0 +1,65 @@
+import { IUser } from '../../user.model';
+import { Observable, of } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { LuUserDisplayPipe, LuDisplayFullname, LuDisplayInitials } from '../../display/index';
+import { delay, map, catchError } from 'rxjs/operators';
+
+interface IHomonyms<U extends IUser = IUser> {
+	[key: string]: U[];
+}
+
+export interface ILuUserHomonymsService<U extends IUser = IUser> {
+	extractHomonyms(users: U[]): U[];
+	enrichHomonyms(homonyms: U[]): Observable<U[]>;
+}
+
+export abstract class ALuUserHomonymsService<U extends IUser = IUser> implements ILuUserHomonymsService<U> {
+	abstract extractHomonyms(users: U[]): U[];
+	abstract enrichHomonyms(homonyms: U[]): Observable<U[]>;
+}
+export class LuUserHomonymsService<U extends IUser = IUser> extends ALuUserHomonymsService<U> implements ILuUserHomonymsService<U> {
+	//  temp - match homonyms on initials to get more homonyms match in dev mode
+	// private _format = LuDisplayFullname.lastfirst;
+	private _format = LuDisplayInitials.lastfirst;
+	extractHomonyms(users: U[]): U[] {
+		const namesCount = {} as { [key: string]: number};

```

Comment:
Why don't you store `usersByName :{ [key: string]: U[]}` to avoid re-scanning users to build `homonymsByname`

Reply:
i did at some point but the code just escalated in complexity quickly. right now for this v1 i'll just fetch homonyms department and hope that in 99%+ of the cases it differentiates homonyms

keeping the homonyms grouped by their same name is only interesting if you intent to treat each group of homonyms separately, the way i implemented it, i just dont care.

i have these 10 users
> alice, bob, bob, charles, derrick, ebenezar, frank, frank, frank and gabe

i extract the homonyms

> bob, bob, frank frank and frank

and enrich them with their department

> alice, bob from accounting, bob from sales, charles, derrick, ebenezar, frank from hr, frank from support, frank from sales and gabe

i dont need to keep bob and bob separate from frank frank and frank

i think i dont need homonym,s by name anymore, its a remainder of my first implementation that was crazy complex